source/Background/Keyring.js

```diff
@@ -117,33 +165,40 @@ export const getKeyringHandler = (type, keyring) => ({
   },
   [HANDLER_TYPES.GET_TRANSACTIONS]: async () => {
     const response = await keyring.getTransactions();
-    return recursiveParseBigint(response);
+    return parseTransactions(response);
   },
   [HANDLER_TYPES.GET_ASSETS]: async ({ refresh }) => {
     try {
       if (!keyring?.isUnlocked) return {};
 
       const { wallets, currentWalletId } = await keyring.getState();
-      let assets = wallets?.[currentWalletId]?.assets;
-      const shouldUpdate = assets?.every((asset) => !Number(asset.amount))
-        || assets?.some((asset) => asset.amount === 'Error')
+      let assets = Object.values(wallets?.[currentWalletId]?.assets);
+      const shouldUpdate = Object.values(assets)?.every((asset) => !Number(asset.amount))
+        || Object.values(assets)?.some((asset) => asset.amount === 'Error')
```

Comment:
I'm afraid these kinda changes might break the extension on the first render. But we should do upgrade tests for this

Reply:
I think on the first render we might have the balances broken with the manual refresh button, but after the first update we shouldn't have any issues